services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/security/OAuth2ResourceServerSecurityConfiguration.kt

```diff
@@ -0,0 +1,91 @@
+package fi.hel.haitaton.hanke.security
+
+import fi.hel.haitaton.hanke.HankeError
+import fi.hel.haitaton.hanke.OBJECT_MAPPER
+import io.sentry.Sentry
+import javax.servlet.http.HttpServletResponse
+import mu.KotlinLogging
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.annotation.Bean
+import org.springframework.core.ParameterizedTypeReference
+import org.springframework.http.HttpMethod
+import org.springframework.security.config.annotation.web.builders.HttpSecurity
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter
+import org.springframework.security.config.annotation.web.configurers.oauth2.server.resource.OAuth2ResourceServerConfigurer
+import org.springframework.security.oauth2.core.OAuth2AuthenticatedPrincipal
+import org.springframework.security.oauth2.core.user.DefaultOAuth2User
+import org.springframework.security.oauth2.server.resource.introspection.OpaqueTokenIntrospector
+import org.springframework.web.reactive.function.client.WebClient
+
+private val logger = KotlinLogging.logger {}
+
+/**
+ * OAuth2 Security Configuration.
+ *
+ * @author Josh Cummings
```

Comment:
I this javadoc necessary?

Reply:
Thank you, this was left over from the example I copied as a base.